### DISCUSSION TITLE
How to update the CloudFlare "trusted IPs" in NGINX?

### DISCUSSION THREAD

**Discussion by matheusb-comp** [serversideup/docker-php#201] [Q&A] [answered]
Currently (v2), the CloudFlare IPs used in NGINX's "realip module" seem to be [hardcoded](https://github.com/serversideup/docker-php/blob/290e14f973572ef1ad0a25acb43a88d1dfc19d1e/src/fpm-nginx/etc/nginx/server-opts.d/remoteip.conf).
To prevent a security issue in case CloudFlare changes their IP ranges, there should be some process to update the file above regulary ([this project](https://github.com/ergin/nginx-cloudflare-real-ip), for example).

Is there a "proper way" to do this? Or should I just customize the `removeip.conf` file using Docker's `bind mount`?

**Comment by jaydrogers** [comment, marked as answer]
I am subscribed to their email list for any IP address changes.

If there was a change, I would push a new image immediately because I use CloudFlare for everything.

Feel free to extend the image as you wish, but for now I wouldn't suggest any changes to the project unless there was larger demand or a larger risk of issues being created.

I feel pretty confident with the email subscription because the IP changes would very unlikely happen overnight. I think I remember this happening once or twice since I have been monitoring it.
